Add session info printing to geopmaccess

bgeltz opened the pull request:
- Fix issues stemming from #2356.
- Add `geopmaccess -s` option to print information about active sessions.
- Fix integration tests that are related to killing the batch session (user owned) and the handling of the batch server (root owned).

bgeltz committed:
Add session info printing to geopmaccess

- Also fix the corresponding integration test.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo commented on service/geopmdpy/access.py:
```diff
@@ -314,9 +314,24 @@ def run(self, is_write, is_all, is_control, group, is_default, is_delete,
         output = None
+        # Determine if user provided -s option
+        if session_pid is not None:
+            file_path = f'/run/geopm-service/session-{session_pid}.json'
+            try:
+                with open(file_path) as fid:
+                    # Early return if session info is requested
+                    return fid.read()
+            except PermissionError as ex:
+                sys.stderr.write(f'Error: {ex}\n\n')
+                return
+            except FileNotFoundError as ex:
+                sys.stderr.write(f'Error: {session_pid} is invalid\n\n')
+                return
+
```
```suggestion
            file_path = f'/run/geopm-service/session-{session_pid}.json'
            result = None
            try:
                with open(file_path) as fid:
                    result = fid.read()
            except PermissionError as ex:
                sys.stderr.write(f'Error: {ex}\n\n')
            except FileNotFoundError as ex:
                sys.stderr.write(f'Error: {session_pid} is invalid\n\n
            # Early return if session info is requested
            return result

```

cmcantalupo replied:
maybe just one early return :)

cmcantalupo committed:
Use existing handling for RuntimeError

- This ensures that return code from geopmaccess CLI relects
  the new error conditions.
- Also enables debug printing of stack when GEOPM_DEBUG is set
  in environment.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
